---
title: "13_PopGen_MaleGenes.rmd"
author: "Garett Slater"
date: "April 21 2022"
output:
  html_document: default
  pdf_document: default
---


```{r knitr setup, include=FALSE,  eval=TRUE, echo=FALSE, warning=FALSE}
library(knitr)
knitr::opts_chunk$set(eval=TRUE, cache=FALSE, message=FALSE, warning=FALSE, 
                      comment = "")
```


```{r data}
setwd("~/OneDrive/Manuscripts/GenesofDrones/WGCNA")
# Display the current working directory
getwd();

library(pacman)
p_load(readxl,dplyr,tidyverse,lme4,nadiv,MCMCglmm,QGglmm,ggpubr,tximport,DESeq2)


#Upload Trait Files
GB_to_NCBI<-read.csv("~/OneDrive/Manuscripts/GenesOfDrones/Resources/GB_to_NCBI.csv")
Sperm<-read_excel("~/OneDrive/Manuscripts/GenesOfDrones/Resources/Expression_Drone_Trait.xlsx",sheet="Sperm")
Weight<-read_excel("~/OneDrive/Manuscripts/GenesOfDrones/Resources/Expression_Drone_Trait.xlsx",sheet="Weight")
Morpho<-read_excel("~/OneDrive/Manuscripts/GenesOfDrones/Resources/Expression_Drone_Trait.xlsx",sheet="Morpho")
ID<-read_excel("~/OneDrive/Manuscripts/GenesOfDrones/Resources/Expression_Drone_Trait.xlsx",sheet="ID")
RFID_out<-read.csv("~/OneDrive/Manuscripts/GenesOfDrones/Resources/RFID_2020Final_out.csv")
RFID_out<-merge(RFID_out,ID,by="UID")
RFID_out$UTCTime_Final<- with(RFID_out, as.POSIXct(paste(UTCDate, UTCTime), format="%m/%d/%Y %H:%M:%OS"))
var_out<-RFID_out %>% group_by(UID) %>% dplyr::summarise(N_flights=length(flight),mean_flight_timediff=mean(timediff)/60,first_flight_age=min(Drone_Age))


#Graph Function
graph_all=function(sum,xaxis,yaxis) {
  ggplot(sum) +
    geom_bar(aes(x=Trait, y=mean), stat="identity", alpha=1,width=0.8,position=position_dodge(2)) +
    geom_errorbar(aes(x=Trait,ymin=mean-se, ymax=mean+se), position=position_dodge(1),width=0, size=1) +
    xlab(xaxis) +
    ylab(yaxis) +
    scale_colour_hue(name="Tissue Type", l=40) + 
    scale_y_continuous(expand = c(0, 0))+
    theme_classic() +
    theme(axis.ticks.x = element_blank(),
          axis.text.x   = element_text(size=10, color="black"),
          axis.title.y  = element_text(face="bold", size=12),
          axis.text.y   = element_text(size=10, color="black"),
          legend.background = element_rect(size=0.5, linetype="solid",colour ="black")) 
  
}


#Sample File and Filtering
Sample<-read_excel("~/OneDrive/Manuscripts/GenesOfDrones/Resources/Expression_Drone_Trait.xlsx",sheet="Sample")
SampleT<-Sample[which(Sample$Tissue=="Testes"),]
#GB to Transcript File for Kallisto
tx2gene=read.csv("~/OneDrive/Manuscripts/GenesOfDrones/Resources/GB_to_NCBI.csv")
Sperm<-read_excel("~/OneDrive/Manuscripts/GenesOfDrones/Resources/Expression_Drone_Trait.xlsx",sheet="Sperm")
ID<-read_excel("~/OneDrive/Manuscripts/GenesOfDrones/Resources/Expression_Drone_Trait.xlsx",sheet="ID")
SampleT<-merge(SampleT,ID)
SampleT<-SampleT%>%distinct(ID2,.keep_all = TRUE)

```

# Sperm Connected Genes

```{r echo=FALSE}
setwd("~/OneDrive/Manuscripts/GenesofDrones/WGCNA")
#Connectivity Testes Module
MM_T<-read.csv("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/MM_T.csv")
colnames(MM_T)[1]<-"GB"

#Black Module
Black<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_T/7.txt")
colnames(Black)[1]<-"GB"
Black_con<-MM_T[,c(1,10)]
###High Connectivity
Black_f<-merge(Black,Black_con)
Black_high<-Black_f[which(Black_f$MM.7>0.9),]
Black_high$Module<-"Black"
Black_high$Trait<-"Sperm-High"
colnames(Black_high)[2]<-"connectivity"

###Low Connectivity
Black_low<-Black_f%>%dplyr::top_n(-30,MM.7)
Black_low$Module<-"Black"
Black_low$Trait<-"Sperm-Low"
colnames(Black_low)[2]<-"connectivity"

#Turquoise Module
Turquoise<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_T/1.txt")
colnames(Turquoise)[1]<-"GB"
Turquoise_con<-MM_T[,c(1,2)]

###Get Connectivity
Turquoise_f<-merge(Turquoise,Turquoise_con)
Turquoise_high<-Turquoise_f[which(Turquoise_f$MM.1>0.9),]
Turquoise_high$Module<-"Turquoise"
Turquoise_high$Trait<-"Sperm-High"
colnames(Turquoise_high)[2]<-"connectivity"

###Low Connectivity
Turquoise_low<-Turquoise_f%>%dplyr::top_n(-30,MM.1)
Turquoise_low$Module<-"Turquoise"
Turquoise_low$Trait<-"Sperm-Low"
colnames(Turquoise_low)[2]<-"connectivity"
```

# Flight Connected Genes

```{r echo=FALSE}
setwd("~/OneDrive/Manuscripts/GenesofDrones/WGCNA")
#Connectivity Brain Module
MM_B<-read.csv("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/MM_B.csv")
colnames(MM_B)[1]<-"GB"

#LightGreen
LightGreen<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_B/18.txt")
colnames(LightGreen)[1]<-"GB"
LightGreen_con<-MM_B[,c(1,19)]
###Get Connectivity
LightGreen_f<-merge(LightGreen,LightGreen_con)
LightGreen_high<-LightGreen_f[which(LightGreen_f$MM.18>0.9),]
LightGreen_high$Module<-"LightGreen"
LightGreen_high$Trait<-"Maturity-High"
colnames(LightGreen_high)[2]<-"connectivity"

###Low Connectivity
LightGreen_low<-LightGreen_f%>%dplyr::top_n(-30,MM.18)
LightGreen_low$Module<-"Turquoise"
LightGreen_low$Trait<-"Maturity-Low"
colnames(LightGreen_low)[2]<-"connectivity"

#Turquoise
Turquoise_2<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_B/1.txt")
colnames(Turquoise_2)[1]<-"GB"
Turquoise_2_con<-MM_B[,c(1,14)]
###Get Connectivity
Turquoise_2_f<-merge(Turquoise_2,Turquoise_2_con)
Turquoise_2_high<-Turquoise_2_f[which(Turquoise_2_f$MM.1>0.9),]
Turquoise_2_high$Module<-"Turquoise_2"
Turquoise_2_high$Trait<-"Maturity-High"
colnames(Turquoise_2_high)[2]<-"connectivity"

###Low Connectivity
Turquoise_2_low<-Turquoise_2_f%>%dplyr::top_n(-30,MM.1)
Turquoise_2_low$Module<-"Turquoise"
Turquoise_2_low$Trait<-"Maturity-Low"
colnames(Turquoise_2_low)[2]<-"connectivity"


##############
# Flight
#############

#Connectivity Brain Module
MM_B<-read.csv("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/MM_B.csv")
colnames(MM_B)[1]<-"GB"

#LightGreen
LightGreen<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_B/18.txt")
colnames(LightGreen)[1]<-"GB"
LightGreen_con<-MM_B[,c(1,19)]
###Get Connectivity
LightGreen_f<-merge(LightGreen,LightGreen_con)
LightGreen_high<-LightGreen_f[which(LightGreen_f$MM.18>0.9),]
LightGreen_high$Module<-"LightGreen"
LightGreen_high$Trait<-"Maturity-High"
colnames(LightGreen_high)[2]<-"connectivity"

###Low Connectivity
LightGreen_low<-LightGreen_f%>%dplyr::top_n(-30,MM.18)
LightGreen_low$Module<-"Turquoise"
LightGreen_low$Trait<-"Maturity-Low"
colnames(LightGreen_low)[2]<-"connectivity"


```

# Weight Connected Genes

```{r echo=FALSE}
setwd("~/OneDrive/Manuscripts/GenesofDrones/WGCNA")
#Connectivity Brain Module
MM_B<-read.csv("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/MM_B.csv")
colnames(MM_B)[1]<-"GB"

#Red
Red<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_B/6.txt")
colnames(Red)[1]<-"GB"
Red_con<-MM_B[,c(1,16)]
###Get Connectivity
Red_f<-merge(Red,Red_con)
Red_high<-Red_f[which(Red_f$MM.6>0.9),]
Red_high$Module<-"Red"
Red_high$Trait<-"Weight-High"
colnames(Red_high)[2]<-"connectivity"

###Low Connectivity
Red_low<-Red_f%>%dplyr::top_n(-30,MM.6)
Red_low$Module<-"Turquoise"
Red_low$Trait<-"Weight-Low"
colnames(Red_low)[2]<-"connectivity"

#Blue
Blue<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_B/2.txt")
colnames(Blue)[1]<-"GB"
Blue_con<-MM_B[,c(1,4)]
###Get Connectivity
Blue_f<-merge(Blue,Blue_con)
Blue_high<-Blue_f[which(Blue_f$MM.2>0.9),]
Blue_high$Module<-"Blue"
Blue_high$Trait<-"Weight-High"
colnames(Blue_high)[2]<-"connectivity"

###Low Connectivity
Blue_low<-Blue_f%>%dplyr::top_n(-30,MM.2)
Blue_low$Module<-"Turquoise"
Blue_low$Trait<-"Weight-Low"
colnames(Blue_low)[2]<-"connectivity"
```

# Pop Gen Analysis
```{r echo=FALSE}
setwd("~/OneDrive/Manuscripts/GenesofDrones/WGCNA")
GL<-rbind(Black_high,Black_low,Turquoise_high,Turquoise_low,Turquoise_2_high,Turquoise_2_low,LightGreen_high,LightGreen_low,Blue_low,Blue_high,Red_low,Red_high)

#PopGene-Gamma
gamma<-read.csv("~/OneDrive/Manuscripts/Published!/Haploid and Sexual Selection Shape the Rate of Evolution of Genes Across the Honey bee (Apis mellifera L.) Genome/Resources/gamma.csv")
GL_gam<-merge(GL,gamma)

sum = GL_gam[!is.na(GL_gam$gamma), ] %>% group_by(Trait) %>% dplyr::summarize(mean=mean(gamma),sd=sd(gamma),N=length(gamma),se=sd/sqrt(N),na.rm=TRUE)
gam_g<-graph_all(sum,"Trait","Gamma")
ff<-filter(GL_gam,Trait=="Maturity-Low"| Trait=="Maturity-High" )
ff<-filter(GL_gam,Trait=="Sperm-Low"| Trait=="Sperm-High" )
ff<-filter(GL_gam,Trait=="Weight-Low"| Trait=="Weight-High" )
ff<-filter(GL_gam,Trait=="Weight-High"| Trait=="Maturity-High"| Trait=="Sperm-High")
f<-aov(gamma~Trait,data=ff)
summary(f)
TukeyHSD(f)



#PopGene-Diversity
pi<-read.csv("~/OneDrive/Manuscripts/Published!/Haploid and Sexual Selection Shape the Rate of Evolution of Genes Across the Honey bee (Apis mellifera L.) Genome/PopGenResources/AMEL_Pi_N_S_ALL_SUM.csv")
pi<-merge(pi,GB_to_NCBI)
pi<-pi[!duplicated(pi$GB), ]
GL_pi<-merge(GL,pi)

##piN
sum = GL_pi[!is.na(GL_pi$piN), ] %>% group_by(Trait) %>% dplyr::summarize(mean=mean(piN),sd=sd(piN),N=length(piN),se=sd/sqrt(N),na.rm=TRUE)
piN_g<-graph_all(sum,"Trait","piN")
ff<-filter(GL_pi,Trait=="Maturity-Low"| Trait=="Maturity-High" )
ff<-filter(GL_pi,Trait=="Sperm-Low"| Trait=="Sperm-High" )
ff<-filter(GL_pi,Trait=="Weight-Low"| Trait=="Weight-High" )
ff<-filter(GL_pi,Trait=="Weight-Low"| Trait=="Maturity-Low"| Trait=="Sperm-Low")
f<-aov(piN~Trait,data=ff)
summary(f)
TukeyHSD(f)



###piS
sum = GL_pi[!is.na(GL_pi$piS), ] %>% group_by(Trait) %>% dplyr::summarize(mean=mean(piS),sd=sd(piS),N=length(piS),se=sd/sqrt(N),na.rm=TRUE)
piS_g<-graph_all(sum,"Trait","piS")
ff<-filter(GL_pi,Trait=="Maturity-Low"| Trait=="Maturity-High" )
ff<-filter(GL_pi,Trait=="Sperm-Low"| Trait=="Sperm-High" )
ff<-filter(GL_pi,Trait=="Weight-Low"| Trait=="Weight-High" )
ff<-filter(GL_pi,Trait=="Weight-High"| Trait=="Maturity-High"| Trait=="Sperm-High")
f<-aov(piS~Trait,data=ff)
summary(f)
TukeyHSD(f)


###piNpiS
sum = GL_pi[!is.na(GL_pi$piNpiS), ] %>% group_by(Trait) %>% dplyr::summarize(mean=mean(piNpiS),sd=sd(piNpiS),N=length(piNpiS),se=sd/sqrt(N),na.rm=TRUE)
piNpiS_g<-graph_all(sum,"Trait","piNpiS")
ff<-filter(GL_pi,Trait=="Sperm-Low"| Trait=="Sperm-High" )
f<-aov(piNpiS~Trait,data=ff)
summary(f)
TukeyHSD(f)


pdf(file = "PopGen_Connectivity.pdf", width = 12, height = 9);
ggarrange(piN_g,piS_g,piNpiS_g,gam_g)
dev.off()
ggarrange(piN_g,piS_g,piNpiS_g,gam_g)
```


# Connectivity based on DEG - Testes
```{r echo=FALSE}
setwd("~/OneDrive/Manuscripts/GenesofDrones/Abundance/")

# Contract Function
contrast = function(type,x,y) {
  lar_d <- data.frame(results(dds, contrast=c(type,x,y)))
  lar_d<-lar_d[which(lar_d$pvalue<0.05),]
  lar_d$GB<-rownames(lar_d)
  lar_d<-lar_d[!duplicated(lar_d$GB), ]
}


contrastNS = function(type,x,y) {
  lar_d <- data.frame(results(dds, contrast=c(type,x,y)))
  lar_d<-lar_d[which(lar_d$pvalue>0.1),]
  lar_d$GB<-rownames(lar_d)
  lar_d<-lar_d[!duplicated(lar_d$GB), ]
}



#DESeq analysis with 
files<-t(SampleT[,2:3])
colnames(files)<-files[2,]
files2<-files[1,]

#Run Kallisto in R. I used Line as the design to account for line differences
txi <- tximport(files2, type = "kallisto", tx2gene = tx2gene)
dds <- DESeqDataSetFromTximport(txi, SampleT, design=~Line)
dds <- dds[rowSums(counts(dds)) > 10, ]
dds <- DESeq(dds)

#Line
#California
CAvP <- contrast("Line","CA","Purdue")
CAvTX <- contrast("Line","CA","Texas")
CAvR <- contrast("Line","CA","Russian")
CA<-merge(CAvP[,c("GB"),drop=FALSE],CAvTX[,c("GB"),drop=FALSE], by="GB")
CA<-merge(CA,CAvR[,c("GB"),drop=FALSE], by="GB")

#Texas
TXvP <- contrast("Line","Texas","Purdue")
TXvCA <- contrast("Line","Texas","CA")
TXvR <- contrast("Line","Texas","Russian")
TX<-merge(TXvP[,c("GB"),drop=FALSE],TXvCA[,c("GB"),drop=FALSE], by="GB")
TX<-merge(TX,TXvR[,c("GB"),drop=FALSE], by="GB")

#Russian
RvP <- contrast("Line","Russian","Purdue")
RvCA <- contrast("Line","Russian","CA")
RvTX <- contrast("Line","Russian","Texas")
R<-merge(RvP[,c("GB"),drop=FALSE],RvCA[,c("GB"),drop=FALSE], by="GB")
R<-merge(R,RvTX[,c("GB"),drop=FALSE], by="GB")

#Purdue
PvR <- contrast("Line","Purdue","Russian")
PvCA <- contrast("Line","Purdue","CA")
PvTX <- contrast("Line","Purdue","Texas")
P<-merge(PvR[,c("GB"),drop=FALSE],PvCA[,c("GB"),drop=FALSE], by="GB")
P<-merge(P,PvTX[,c("GB"),drop=FALSE], by="GB")

#Line DEGs
LN<-rbind(CA,R,TX,P)
LN<-distinct(LN)
LN$Type<-"S"

#Line
#California
CAvP <- contrastNS("Line","CA","Purdue")
CAvTX <- contrastNS("Line","CA","Texas")
CAvR <- contrastNS("Line","CA","Russian")
CA<-merge(CAvP[,c("GB"),drop=FALSE],CAvTX[,c("GB"),drop=FALSE], by="GB")
CA<-merge(CA,CAvR[,c("GB"),drop=FALSE], by="GB")

#Texas
TXvP <- contrastNS("Line","Texas","Purdue")
TXvCA <- contrastNS("Line","Texas","CA")
TXvR <- contrastNS("Line","Texas","Russian")
TX<-merge(TXvP[,c("GB"),drop=FALSE],TXvCA[,c("GB"),drop=FALSE], by="GB")
TX<-merge(TX,TXvR[,c("GB"),drop=FALSE], by="GB")

#Russian
RvP <- contrastNS("Line","Russian","Purdue")
RvCA <- contrastNS("Line","Russian","CA")
RvTX <- contrastNS("Line","Russian","Texas")
R<-merge(RvP[,c("GB"),drop=FALSE],RvCA[,c("GB"),drop=FALSE], by="GB")
R<-merge(R,RvTX[,c("GB"),drop=FALSE], by="GB")

#Purdue
PvR <- contrastNS("Line","Purdue","Russian")
PvCA <- contrastNS("Line","Purdue","CA")
PvTX <- contrastNS("Line","Purdue","Texas")
P<-merge(PvR[,c("GB"),drop=FALSE],PvCA[,c("GB"),drop=FALSE], by="GB")
P<-merge(P,PvTX[,c("GB"),drop=FALSE], by="GB")

#Line DEGs
LN_NS<-rbind(CA,R,TX,P)
LN_NS<-distinct(LN_NS)
LN_NS$Type<-"NS"

LN<-rbind(LN,LN_NS)

setwd("~/OneDrive/Manuscripts/GenesofDrones/WGCNA")
#Connectivity Testes Module
MM_T<-read.csv("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/MM_T.csv")
colnames(MM_T)[1]<-"GB"

#Black Module
Black<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_T/7.txt")
colnames(Black)[1]<-"GB"
Black_con<-MM_T[,c(1,10)]
Black_f<-merge(Black,Black_con)
colnames(Black_f)[2]<-"MM"
Black_f$Trait<-"Sperm"

#Turquoise Module
Turquoise<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_T/1.txt")
colnames(Turquoise)[1]<-"GB"
Turquoise_con<-MM_T[,c(1,2)]
Turquoise_f<-merge(Turquoise,Turquoise_con)
colnames(Turquoise_f)[2]<-"MM"
Turquoise_f$Trait<-"Sperm"

#Sperm Modules
Sperm_MM<-rbind(Black_f,Turquoise_f)

#Connectivity
LN<-merge(LN,Sperm_MM,by="GB")


aov<-aov(MM~Type,data=LN)
summary(aov)

LN%>%group_by(Type)%>%summarize(mean=mean(MM))

```


# Connectivity based on DEG - Brain
```{r echo=FALSE}
setwd("~/OneDrive/Manuscripts/GenesofDrones/Abundance/")
Sample<-read_excel("~/OneDrive/Manuscripts/GenesOfDrones/Resources/Expression_Drone_Trait.xlsx",sheet="Sample")
SampleB<-Sample[which(Sample$Tissue=="Brain"),]
SampleB<-merge(SampleB,ID)
SampleB<-SampleB%>%distinct(ID2,.keep_all = TRUE)

# Contract Function
contrast = function(type,x,y) {
  lar_d <- data.frame(results(dds, contrast=c(type,x,y)))
  lar_d<-lar_d[which(lar_d$pvalue<0.05),]
  lar_d$GB<-rownames(lar_d)
  lar_d<-lar_d[!duplicated(lar_d$GB), ]
}


contrastNS = function(type,x,y) {
  lar_d <- data.frame(results(dds, contrast=c(type,x,y)))
  lar_d<-lar_d[which(lar_d$pvalue>0.1),]
  lar_d$GB<-rownames(lar_d)
  lar_d<-lar_d[!duplicated(lar_d$GB), ]
}

files<-t(SampleB[,2:3])
colnames(files)<-files[2,]
files2<-files[1,]

#Run Kallisto
txi <- tximport(files2, type = "kallisto", tx2gene = tx2gene)
dds <- DESeqDataSetFromTximport(txi, SampleB, design=~Line)
dds <- dds[rowSums(counts(dds)) > 10, ]
dds <- DESeq(dds)


#Line
#California
CAvP <- contrast("Line","CA","Purdue")
CAvTX <- contrast("Line","CA","Texas")
CAvR <- contrast("Line","CA","Russian")
CA<-merge(CAvP[,c("GB"),drop=FALSE],CAvTX[,c("GB"),drop=FALSE], by="GB")
CA<-merge(CA,CAvR[,c("GB"),drop=FALSE], by="GB")

#Texas
TXvP <- contrast("Line","Texas","Purdue")
TXvCA <- contrast("Line","Texas","CA")
TXvR <- contrast("Line","Texas","Russian")
TX<-merge(TXvP[,c("GB"),drop=FALSE],TXvCA[,c("GB"),drop=FALSE], by="GB")
TX<-merge(TX,TXvR[,c("GB"),drop=FALSE], by="GB")

#Russian
RvP <- contrast("Line","Russian","Purdue")
RvCA <- contrast("Line","Russian","CA")
RvTX <- contrast("Line","Russian","Texas")
R<-merge(RvP[,c("GB"),drop=FALSE],RvCA[,c("GB"),drop=FALSE], by="GB")
R<-merge(R,RvTX[,c("GB"),drop=FALSE], by="GB")

#Purdue
PvR <- contrast("Line","Purdue","Russian")
PvCA <- contrast("Line","Purdue","CA")
PvTX <- contrast("Line","Purdue","Texas")
P<-merge(PvR[,c("GB"),drop=FALSE],PvCA[,c("GB"),drop=FALSE], by="GB")
P<-merge(P,PvTX[,c("GB"),drop=FALSE], by="GB")

#Line DEGs
LN_B<-rbind(CA,R,TX,P)
LN_B<-distinct(LN_B)
LN_B$Type<-"S"

#Line
#California
CAvP <- contrastNS("Line","CA","Purdue")
CAvTX <- contrastNS("Line","CA","Texas")
CAvR <- contrastNS("Line","CA","Russian")
CA<-merge(CAvP[,c("GB"),drop=FALSE],CAvTX[,c("GB"),drop=FALSE], by="GB")
CA<-merge(CA,CAvR[,c("GB"),drop=FALSE], by="GB")

#Texas
TXvP <- contrastNS("Line","Texas","Purdue")
TXvCA <- contrastNS("Line","Texas","CA")
TXvR <- contrastNS("Line","Texas","Russian")
TX<-merge(TXvP[,c("GB"),drop=FALSE],TXvCA[,c("GB"),drop=FALSE], by="GB")
TX<-merge(TX,TXvR[,c("GB"),drop=FALSE], by="GB")

#Russian
RvP <- contrastNS("Line","Russian","Purdue")
RvCA <- contrastNS("Line","Russian","CA")
RvTX <- contrastNS("Line","Russian","Texas")
R<-merge(RvP[,c("GB"),drop=FALSE],RvCA[,c("GB"),drop=FALSE], by="GB")
R<-merge(R,RvTX[,c("GB"),drop=FALSE], by="GB")

#Purdue
PvR <- contrastNS("Line","Purdue","Russian")
PvCA <- contrastNS("Line","Purdue","CA")
PvTX <- contrastNS("Line","Purdue","Texas")
P<-merge(PvR[,c("GB"),drop=FALSE],PvCA[,c("GB"),drop=FALSE], by="GB")
P<-merge(P,PvTX[,c("GB"),drop=FALSE], by="GB")

#Line DEGs
LN_NS<-rbind(CA,R,TX,P)
LN_NS<-distinct(LN_NS)
LN_NS$Type<-"NS"

LN_B<-rbind(LN_B,LN_NS)

setwd("~/OneDrive/Manuscripts/GenesofDrones/WGCNA")
#Connectivity Brain Module
MM_B<-read.csv("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/MM_B.csv")
colnames(MM_B)[1]<-"GB"

#Red
Red<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_B/6.txt")
colnames(Red)[1]<-"GB"
Red_con<-MM_B[,c(1,16)]
Red_f<-merge(Red,Red_con)
colnames(Red_f)[2]<-"MM"
Red_f$Trait<-"Weight"

#Blue
Blue<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_B/2.txt")
colnames(Blue)[1]<-"GB"
Blue_con<-MM_B[,c(1,4)]
Blue_f<-merge(Blue,Blue_con)
colnames(Blue_f)[2]<-"MM"
Blue_f$Trait<-"Weight"

#LightGreen
LightGreen<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_B/18.txt")
colnames(LightGreen)[1]<-"GB"
LightGreen_con<-MM_B[,c(1,19)]
LightGreen_f<-merge(LightGreen,LightGreen_con)
colnames(LightGreen_f)[2]<-"MM"
LightGreen_f$Trait<-"Maturity"


#Turquoise
Turquoise_2<-read.table("~/OneDrive/Manuscripts/GenesofDrones/WGCNA/Module_B/1.txt")
colnames(Turquoise_2)[1]<-"GB"
Turquoise_2_con<-MM_B[,c(1,14)]
Turquoise_2_f<-merge(Turquoise_2,Turquoise_2_con)
colnames(Turquoise_2_f)[2]<-"MM"
Turquoise_2_f$Trait<-"Maturity"

Brain_MM<-rbind(Turquoise_2_f,LightGreen_f,Red_f,Blue_f)
#Connectivity
LN_B<-merge(LN_B,Brain_MM,by="GB")

#Final
Ln_final<-rbind(LN,LN_B)
x<-Ln_final[which(Ln_final$Trait=="Weight"),]
aov<-aov(MM~Type,data=x)
summary(aov)





sum = Ln_final %>% group_by(Trait,Type) %>% dplyr::summarize(mean=mean(MM),sd=sd(MM),N=length(MM),se=sd/sqrt(N),na.rm=TRUE)
MM_g<-graph_all(sum,"Trait","MM")

  ggplot(data=sum,aes(x=Trait, y=mean, fill=Type)) +
    geom_bar( stat="identity", alpha=1,width=0.8,position=position_dodge()) +
    geom_errorbar(aes(ymin=mean-se,ymax=mean+se), width=0, size=1,position=position_dodge(.9)) +
    xlab("Trait") +
    ylab("Connectivity") +
    scale_colour_hue(name="Tissue Type", l=40) + 
    scale_y_continuous(expand = c(0, 0))+
    theme_classic() +
    theme(axis.ticks.x = element_blank(),
          axis.text.x   = element_text(size=10, color="black"),
          axis.title.y  = element_text(face="bold", size=12),
          axis.text.y   = element_text(size=10, color="black"),
          legend.background = element_rect(size=0.5, linetype="solid",colour ="black")) 

x<-Ln_final[which(Ln_final$Trait=="Weight"),]
aov<-aov(MM~Type,data=x)
summary(aov)



```

